### PR TITLE
Improve docs of n_step_lstm

### PR DESCRIPTION
**Pull request by keisuke-umezawa**
This is PR for improving docs of functions/links. Related issue: #2182

Modified functions/links:

- functions/connection/n_step_lstm.py

### CODE REVIEW

File: chainer/functions/connection/n_step_lstm.py [hunk 1]
```diff
@@ -79,40 +79,41 @@ def n_step_lstm(
...
+        bs (list of list of :class:`~chainer.Variable`): Bias vectors.
+            ``bs[i]`` represnents biases for i-th layer.
             Each ``bs[i]`` is a list containing eight vectors.
-            ``bs[i][j]`` is corresponding with ``b_j`` in the equation.
-            Shape of each matrix is ``(N,)`` where ``N`` is dimention of
+            ``bs[i][j]`` is corresponding with :math:`b_j` in the equation.
+            Shape of each matrix is ``(N,)`` where ``N`` is dimension of
             hidden units.
-        xs (list of chainer.Variable): A list of :class:`~chainer.Variable`
+        xs (list of :class:`~chainer.Variable`):
```
Comment (niboshi) [inline chainer/functions/connection/n_step_lstm.py]: * ... is the mini-batch size for ...
* Typo: sequneces -> sequences
* This sentence is not clear. Could you write as the following? 

>~Note that this functions supports variable length sequences.~ (delete this)
>The sequences must be transposed. :func:`~chainer.functions.transpose_sequence` can be used to transpose a list of :func:`~chainer.Variable`\ s each representing a sequence.
>When the sequences have different lengths, they must be sorted in descending order of their lengths before transposing.

File: chainer/functions/connection/n_step_lstm.py [hunk 2]
```diff
@@ -194,49 +237,100 @@ def n_step_bilstm(
...
+            direction. (``m == 0`` means the forward direction and ``m == 1``
+            means the backward direction.)
             Each ``bs[i]`` is a list containing eight vectors.
-            ``bs[i][j]`` is corresponding with ``b_j`` in the equation.
-            Shape of each matrix is ``(N,)`` where ``N`` is dimention of
-            hidden units.
-        xs (list of chainer.Variable): A list of :class:`~chainer.Variable`
+            ``bs[i][j]`` is corresponding with :math:`b_j` in the equation.
+            Shape of each matrix is ``(N,)``.
+        xs (list of :class:`~chainer.Variable`):
```
Comment (niboshi) [inline chainer/functions/connection/n_step_lstm.py]: Ditto.
* ... is the mini-batch size for ...
* Typo: sequneces -> sequences
* This sentence is not clear. Could you write as the following? 

>~Note that this functions supports variable length sequences.~ (delete this)
>The sequences must be transposed. :func:`~chainer.functions.transpose_sequence` can be used to transpose a list of :func:`~chainer.Variable`\ s each representing a sequence.
>When the sequences have different lengths, they must be sorted in descending order of their lengths before transposing.

File: chainer/functions/connection/n_step_lstm.py [hunk 3]
```diff
@@ -168,17 +210,18 @@ def n_step_bilstm(
...
 
     where :math:`W^{f}` is weight matrices for forward-LSTM, :math:`W^{b}` is
     weight matrices for backward-LSTM.
 
     As the function accepts a sequence, it calculates :math:`h_t` for all
     :math:`t` with one call. Eight weight matrices and eight bias vectors are
-    required for each layers. So, when :math:`S` layers exists, you need to
-    prepare :math:`8S` weigth matrices and :math:`8S` bias vectors.
+    required for each layer of each direction. So, when :math:`S` layers
+    exists, you need to prepare :math:`16S` weigth matrices and :math:`16S`
```
Comment (niboshi) [inline chainer/functions/connection/n_step_lstm.py]: exists -> exist

File: chainer/functions/connection/n_step_lstm.py [hunk 4]
```diff
@@ -194,49 +237,100 @@ def n_step_bilstm(
...
-        bs (list of list of chainer.Variable): Bias vectors. ``bs[i]``
-            represnents biases for i-th layer.
+        ws (list of list of :class:`~chainer.Variable`): Weight matrices.
+            ``ws[2 * l + m]`` represents weights for l-th layer of m-th
+            direction. (``m == 0`` means the forward direction and ``m == 1``
+            means the backward direction.) Each ``ws[i]`` is a list containing
+            eight matrices. ``ws[i][j]`` is corresponding with :math:`W_j` in
+            the equation. ``ws[0][j]`` and ``ws[1][j]`` where ``0 <= j < 4`` is
+            ``(I, N)`` shape because they are multiplied with input variables,
+            where ``I`` is size of input. ``ws[i][j]``  where ``2 <= i`` and
```
Comment (niboshi) [inline chainer/functions/connection/n_step_lstm.py]: is the size of the input

File: chainer/functions/connection/n_step_lstm.py [hunk 5]
```diff
@@ -194,49 +237,100 @@ def n_step_bilstm(
...
-            Only ``ws[0][j]`` where ``0 <= j < 4`` is ``(I, N)`` shape as they
-            are multiplied with input variables. All other matrices has
-            ``(N, N)`` shape.
-        bs (list of list of chainer.Variable): Bias vectors. ``bs[i]``
-            represnents biases for i-th layer.
+        ws (list of list of :class:`~chainer.Variable`): Weight matrices.
+            ``ws[2 * l + m]`` represents weights for l-th layer of m-th
+            direction. (``m == 0`` means the forward direction and ``m == 1``
+            means the backward direction.) Each ``ws[i]`` is a list containing
+            eight matrices. ``ws[i][j]`` is corresponding with :math:`W_j` in
```
Comment (niboshi) [inline chainer/functions/connection/n_step_lstm.py]: ws[i][j] corresponds to W_j

File: chainer/functions/connection/n_step_lstm.py [hunk 6]
```diff
@@ -194,49 +237,100 @@ def n_step_bilstm(
...
+            the equation. ``ws[0][j]`` and ``ws[1][j]`` where ``0 <= j < 4`` is
```
Comment (niboshi) [inline chainer/functions/connection/n_step_lstm.py]: ... is (I, N) shape -> ... are (I, N)-shaped

File: chainer/functions/connection/n_step_lstm.py [hunk 7]
```diff
@@ -194,49 +237,100 @@ def n_step_bilstm(
...
+            ``(I, N)`` shape because they are multiplied with input variables,
+            where ``I`` is size of input. ``ws[i][j]``  where ``2 <= i`` and
+            ``0 <= j < 4`` is ``(N, 2N)`` because they are multiplied with two
```
Comment (niboshi) [inline chainer/functions/connection/n_step_lstm.py]: are (N, 2N)-shaped

File: chainer/functions/connection/n_step_lstm.py [hunk 8]
```diff
@@ -194,49 +237,100 @@ def n_step_bilstm(
...
+            hidden layers :math:`h_t = [h^{f}_t; h^{b}_t]`. All other matrices
+            has ``(N, N)`` shape.
```
Comment (niboshi) [inline chainer/functions/connection/n_step_lstm.py]: are (N, N)-shaped

File: chainer/functions/connection/n_step_lstm.py [hunk 9]
```diff
@@ -194,49 +237,100 @@ def n_step_bilstm(
...
+        bs (list of list of :class:`~chainer.Variable`): Bias vectors.
+            ``bs[2 * l + m]`` represents weights for l-th layer of m-th
+            direction. (``m == 0`` means the forward direction and ``m == 1``
+            means the backward direction.)
             Each ``bs[i]`` is a list containing eight vectors.
-            ``bs[i][j]`` is corresponding with ``b_j`` in the equation.
-            Shape of each matrix is ``(N,)`` where ``N`` is dimention of
-            hidden units.
-        xs (list of chainer.Variable): A list of :class:`~chainer.Variable`
+            ``bs[i][j]`` is corresponding with :math:`b_j` in the equation.
```
Comment (niboshi) [inline chainer/functions/connection/n_step_lstm.py]: bs[i][j] corresponds to b_j

File: chainer/functions/connection/n_step_lstm.py [hunk 10]
```diff
@@ -194,49 +237,100 @@ def n_step_bilstm(
...
+            Shape of each matrix is ``(N,)``.
```
Comment (niboshi) [inline chainer/functions/connection/n_step_lstm.py]: The shape of ...

File: chainer/functions/connection/n_step_lstm.py [hunk 11]
```diff
@@ -194,49 +237,100 @@ def n_step_bilstm(
...
+        ``hy``, ``cy`` and ``ys``.
 
             - ``hy`` is an updated hidden states whose shape is same as ``hx``.
             - ``cy`` is an updated cell states whose shape is same as ``cx``.
             - ``ys`` is a list of :class:`~chainer.Variable` . Each element
               ``ys[t]`` holds hidden states of the last layer corresponding
-              to an input ``xs[t]``. Its shape is ``(B_t, N)`` where ``B_t`` is
-              mini-batch size for time ``t``, and ``N`` is size of hidden
+              to an input ``xs[t]``. Its shape is ``(B_t, 2N)`` where ``B_t``
+              is mini-batch size for time ``t``, and ``N`` is size of hidden
```
Comment (niboshi) [inline chainer/functions/connection/n_step_lstm.py]: * ... is the mini-batch size
* ... N is the size of the hidden units.

File: chainer/functions/connection/n_step_lstm.py [hunk 12]
```diff
@@ -255,50 +349,50 @@ def n_step_lstm_base(
     Args:
         n_layers(int): Number of layers.
         dropout_ratio(float): Dropout ratio.
-        hx (chainer.Variable): Variable holding stacked hidden states.
+        hx (~chainer.Variable): Variable holding stacked hidden states.
             Its shape is ``(S, B, N)`` where ``S`` is number of layers and is
             equal to ``n_layers``, ``B`` is mini-batch size, and ``N`` is
-            dimention of hidden units.
-        cx (chainer.Variable): Variable holding stacked cell states.
+            dimension of hidden units.
```
Comment (niboshi) [inline chainer/functions/connection/n_step_lstm.py]: ... is the dimension of the hidden units.

File: chainer/functions/connection/n_step_lstm.py [hunk 13]
```diff
@@ -255,50 +349,50 @@ def n_step_lstm_base(
...
+        cx (~chainer.Variable): Variable holding stacked cell states.
             It has the same shape as ``hx``.
-        ws (list of list of chainer.Variable): Weight matrices. ``ws[i]``
-            represents weights for i-th layer.
+        ws (list of list of :class:`~chainer.Variable`): Weight matrices.
+            ``ws[i]`` represents weights for i-th layer.
```
Comment (niboshi) [inline chainer/functions/connection/n_step_lstm.py]: ... represents the weights of the i-th layer

File: chainer/functions/connection/n_step_lstm.py [hunk 14]
```diff
@@ -255,50 +349,50 @@ def n_step_lstm_base(
...
             Each ``ws[i]`` is a list containing eight matrices.
-            ``ws[i][j]`` is corresponding with ``W_j`` in the equation.
+            ``ws[i][j]`` is corresponding with :math:`W_j` in the equation.
```
Comment (niboshi) [inline chainer/functions/connection/n_step_lstm.py]: ws[i][j] corresponds to W-j

File: chainer/functions/connection/n_step_lstm.py [hunk 15]
```diff
@@ -255,50 +349,50 @@ def n_step_lstm_base(
...
             Only ``ws[0][j]`` where ``0 <= j < 4`` is ``(I, N)`` shape as they
-            are multiplied with input variables. All other matrices has
-            ``(N, N)`` shape.
-        bs (list of list of chainer.Variable): Bias vectors. ``bs[i]``
-            represnents biases for i-th layer.
+            are multiplied with input variables, where ``I`` is size of input
```
Comment (niboshi) [inline chainer/functions/connection/n_step_lstm.py]: * ... is the size of the input and N is the dimension of the hidden units.
* ... are (N, N)-shaped.

File: chainer/functions/connection/n_step_lstm.py [hunk 16]
```diff
@@ -255,50 +349,50 @@ def n_step_lstm_base(
...
+            and units ``N`` is dimension of hidden units. All other matrices
+            has ``(N, N)`` shape.
+        bs (list of list of :class:`~chainer.Variable`): Bias vectors.
+            ``bs[i]`` represnents biases for i-th layer.
```
Comment (niboshi) [inline chainer/functions/connection/n_step_lstm.py]: * Typo: represents
* ...represents the biases of the i-th layer

File: chainer/functions/connection/n_step_lstm.py [hunk 17]
```diff
@@ -255,50 +349,50 @@ def n_step_lstm_base(
...
             Each ``bs[i]`` is a list containing eight vectors.
-            ``bs[i][j]`` is corresponding with ``b_j`` in the equation.
-            Shape of each matrix is ``(N,)`` where ``N`` is dimention of
-            hidden units.
-        xs (list of chainer.Variable): A list of :class:`~chainer.Variable`
+            ``bs[i][j]`` is corresponding with :math:`b_j` in the equation.
```
Comment (niboshi) [inline chainer/functions/connection/n_step_lstm.py]: bs[i][j] corresponds to b_j

File: chainer/functions/connection/n_step_lstm.py [hunk 18]
```diff
@@ -255,50 +349,50 @@ def n_step_lstm_base(
...
+            Shape of each matrix is ``(N,)``.
+        xs (list of :class:`~chainer.Variable`):
```
Comment (niboshi) [inline chainer/functions/connection/n_step_lstm.py]: Ditto.
Ditto.
* ... is the mini-batch size for ...
* Typo: sequneces -> sequences
* This sentence is not clear. Could you write as the following? 

>~Note that this functions supports variable length sequences.~ (delete this)
>The sequences must be transposed. :func:`~chainer.functions.transpose_sequence` can be used to transpose a list of :func:`~chainer.Variable`\ s each representing a sequence.
>When the sequences have different lengths, they must be sorted in descending order of their lengths before transposing.

File: chainer/functions/connection/n_step_lstm.py [hunk 19]
```diff
@@ -255,50 +349,50 @@ def n_step_lstm_base(
...
+        ``hy``, ``cy`` and ``ys``.
 
             - ``hy`` is an updated hidden states whose shape is same as ``hx``.
             - ``cy`` is an updated cell states whose shape is same as ``cx``.
             - ``ys`` is a list of :class:`~chainer.Variable` . Each element
               ``ys[t]`` holds hidden states of the last layer corresponding
               to an input ``xs[t]``. Its shape is ``(B_t, N)`` where ``B_t`` is
-              mini-batch size for time ``t``, and ``N`` is size of hidden
-              units. Note that ``B_t`` is the same value as ``xs[t]``.
+              mini-batch size for time ``t``. Note that ``B_t`` is the same
```
Comment (niboshi) [inline chainer/functions/connection/n_step_lstm.py]: is the mini-batch size

File: chainer/functions/connection/n_step_lstm.py [hunk 20]
```diff
@@ -60,10 +60,10 @@ def n_step_lstm(
 
     As the function accepts a sequence, it calculates :math:`h_t` for all
     :math:`t` with one call. Eight weight matrices and eight bias vectors are
-    required for each layers. So, when :math:`S` layers exists, you need to
+    required for each layer. So, when :math:`S` layers exists, you need to
```
[the remark this answers is not in the document]
Comment (keisuke-umezawa) [reply chainer/functions/connection/n_step_lstm.py]: I fixed next commit.

File: chainer/functions/connection/n_step_lstm.py [hunk 21]
```diff
@@ -60,10 +60,10 @@ def n_step_lstm(
...
     prepare :math:`8S` weigth matrices and :math:`8S` bias vectors.
 
-    If the number of layers ``n_layers`` is greather than :math:`1`, input
+    If the number of layers ``n_layers`` is greater than :math:`1`, input
```
[the remark this answers is not in the document]
Comment (keisuke-umezawa) [reply chainer/functions/connection/n_step_lstm.py]: I fixed next commit.

File: chainer/functions/connection/n_step_lstm.py [hunk 22]
```diff
@@ -60,10 +60,10 @@ def n_step_lstm(
...
     of ``k``-th layer is hidden state ``h_t`` of ``k-1``-th layer.
     Note that all input variables except first layer may have different shape
```
[the remark this answers is not in the document]
Comment (keisuke-umezawa) [reply chainer/functions/connection/n_step_lstm.py]: I fixed next commit.

File: chainer/functions/connection/n_step_lstm.py [hunk 23]
```diff
@@ -79,40 +79,41 @@ def n_step_lstm(
     Args:
         n_layers(int): Number of layers.
         dropout_ratio(float): Dropout ratio.
-        hx (chainer.Variable): Variable holding stacked hidden states.
+        hx (~chainer.Variable): Variable holding stacked hidden states.
             Its shape is ``(S, B, N)`` where ``S`` is number of layers and is
```
[the remark this answers is not in the document]
Comment (keisuke-umezawa) [reply chainer/functions/connection/n_step_lstm.py]: I  fixed next commit.

File: chainer/functions/connection/n_step_lstm.py [hunk 24]
```diff
@@ -79,40 +79,41 @@ def n_step_lstm(
...
             equal to ``n_layers``, ``B`` is mini-batch size, and ``N`` is
-            dimention of hidden units.
-        cx (chainer.Variable): Variable holding stacked cell states.
+            dimension of hidden units.
```
[the remark this answers is not in the document]
Comment (keisuke-umezawa) [reply chainer/functions/connection/n_step_lstm.py]: I fixed next commit.

File: chainer/functions/connection/n_step_lstm.py [hunk 25]
```diff
@@ -79,40 +79,41 @@ def n_step_lstm(
...
+        cx (~chainer.Variable): Variable holding stacked cell states.
             It has the same shape as ``hx``.
-        ws (list of list of chainer.Variable): Weight matrices. ``ws[i]``
-            represents weights for i-th layer.
+        ws (list of list of :class:`~chainer.Variable`): Weight matrices.
+            ``ws[i]`` represents weights for i-th layer.
```
[the remark this answers is not in the document]
Comment (keisuke-umezawa) [reply chainer/functions/connection/n_step_lstm.py]: I fixed next commit.

File: chainer/functions/connection/n_step_lstm.py [hunk 26]
```diff
@@ -79,40 +79,41 @@ def n_step_lstm(
...
             Each ``ws[i]`` is a list containing eight matrices.
-            ``ws[i][j]`` is corresponding with ``W_j`` in the equation.
+            ``ws[i][j]`` is corresponding with :math:`W_j` in the equation.
```
[the remark this answers is not in the document]
Comment (keisuke-umezawa) [reply chainer/functions/connection/n_step_lstm.py]: I fixed next commit.

File: chainer/functions/connection/n_step_lstm.py [hunk 27]
```diff
@@ -79,40 +79,41 @@ def n_step_lstm(
...
             Only ``ws[0][j]`` where ``0 <= j < 4`` is ``(I, N)`` shape as they
-            are multiplied with input variables. All other matrices has
-            ``(N, N)`` shape.
-        bs (list of list of chainer.Variable): Bias vectors. ``bs[i]``
-            represnents biases for i-th layer.
+            are multiplied with input variables, where ``I`` is size of input
```
[the remark this answers is not in the document]
Comment (keisuke-umezawa) [reply chainer/functions/connection/n_step_lstm.py]: I fixed next commit.

File: chainer/functions/connection/n_step_lstm.py [hunk 28]
```diff
@@ -79,40 +79,41 @@ def n_step_lstm(
...
+            and units ``N`` is dimension of hidden units. All other matrices
+            has ``(N, N)`` shape.
+        bs (list of list of :class:`~chainer.Variable`): Bias vectors.
+            ``bs[i]`` represnents biases for i-th layer.
```
[the remark this answers is not in the document]
Comment (keisuke-umezawa) [reply chainer/functions/connection/n_step_lstm.py]: I fixed next commit.

File: chainer/functions/connection/n_step_lstm.py [hunk 29]
```diff
@@ -79,40 +79,41 @@ def n_step_lstm(
...
             Each ``bs[i]`` is a list containing eight vectors.
-            ``bs[i][j]`` is corresponding with ``b_j`` in the equation.
-            Shape of each matrix is ``(N,)`` where ``N`` is dimention of
+            ``bs[i][j]`` is corresponding with :math:`b_j` in the equation.
```
[the remark this answers is not in the document]
Comment (keisuke-umezawa) [reply chainer/functions/connection/n_step_lstm.py]: I fixed next commit.

File: chainer/functions/connection/n_step_lstm.py [hunk 30]
```diff
@@ -79,40 +79,41 @@ def n_step_lstm(
...
+            Shape of each matrix is ``(N,)`` where ``N`` is dimension of
```
[the remark this answers is not in the document]
Comment (keisuke-umezawa) [reply chainer/functions/connection/n_step_lstm.py]: I fixed next commit.

File: chainer/functions/connection/n_step_lstm.py (hunk 3 above)
Comment (keisuke-umezawa) [reply chainer/functions/connection/n_step_lstm.py]: I fixed next commit.

File: chainer/functions/connection/n_step_lstm.py (hunk 5 above)
Comment (keisuke-umezawa) [reply chainer/functions/connection/n_step_lstm.py]: I fixed next commit.

File: chainer/functions/connection/n_step_lstm.py (hunk 6 above)
Comment (keisuke-umezawa) [reply chainer/functions/connection/n_step_lstm.py]: I fixed next commit.

File: chainer/functions/connection/n_step_lstm.py (hunk 7 above)
Comment (keisuke-umezawa) [reply chainer/functions/connection/n_step_lstm.py]: I fixed next commit.

File: chainer/functions/connection/n_step_lstm.py (hunk 8 above)
Comment (keisuke-umezawa) [reply chainer/functions/connection/n_step_lstm.py]: I fixed next commit.

File: chainer/functions/connection/n_step_lstm.py (hunk 9 above)
Comment (keisuke-umezawa) [reply chainer/functions/connection/n_step_lstm.py]: I fixed next commit.

File: chainer/functions/connection/n_step_lstm.py (hunk 10 above)
Comment (keisuke-umezawa) [reply chainer/functions/connection/n_step_lstm.py]: I fixed next commit.

File: chainer/functions/connection/n_step_lstm.py (hunk 1 above)
Comment (keisuke-umezawa) [reply chainer/functions/connection/n_step_lstm.py]: I fixed next commit.

File: chainer/functions/connection/n_step_lstm.py (hunk 4 above)
Comment (keisuke-umezawa) [reply chainer/functions/connection/n_step_lstm.py]: I fixed next commit.

File: chainer/functions/connection/n_step_lstm.py (hunk 2 above)
Comment (keisuke-umezawa) [reply chainer/functions/connection/n_step_lstm.py]: I fixed next commit.

File: chainer/functions/connection/n_step_lstm.py (hunk 11 above)
Comment (keisuke-umezawa) [reply chainer/functions/connection/n_step_lstm.py]: I fixed next commit.

File: chainer/functions/connection/n_step_lstm.py (hunk 12 above)
Comment (keisuke-umezawa) [reply chainer/functions/connection/n_step_lstm.py]: I fixed next commit.

File: chainer/functions/connection/n_step_lstm.py (hunk 14 above)
Comment (keisuke-umezawa) [reply chainer/functions/connection/n_step_lstm.py]: I fixed next commit.

File: chainer/functions/connection/n_step_lstm.py (hunk 13 above)
Comment (keisuke-umezawa) [reply chainer/functions/connection/n_step_lstm.py]: I fixed next commit.

File: chainer/functions/connection/n_step_lstm.py (hunk 15 above)
Comment (keisuke-umezawa) [reply chainer/functions/connection/n_step_lstm.py]: I fixed next commit.

File: chainer/functions/connection/n_step_lstm.py (hunk 16 above)
Comment (keisuke-umezawa) [reply chainer/functions/connection/n_step_lstm.py]: I fixed next commit.

File: chainer/functions/connection/n_step_lstm.py (hunk 18 above)
Comment (keisuke-umezawa) [reply chainer/functions/connection/n_step_lstm.py]: I fixed next commit.

File: chainer/functions/connection/n_step_lstm.py (hunk 17 above)
Comment (keisuke-umezawa) [reply chainer/functions/connection/n_step_lstm.py]: I fixed next commit.

File: chainer/functions/connection/n_step_lstm.py (hunk 19 above)
Comment (keisuke-umezawa) [reply chainer/functions/connection/n_step_lstm.py]: I fixed next commit.

**Comment by keisuke-umezawa** [comment]
@niboshi Thank you for the review! I followed your review.

File: chainer/functions/connection/n_step_lstm.py [hunk 31]
```diff
@@ -77,55 +77,99 @@ def n_step_lstm(
...
-            order by length, and transpose the sorted sequence.
-            :func:`~chainer.functions.transpose_sequence` transpose a list
-            of :func:`~chainer.Variable` holding sequence.
+            for time ``t``. Its shape is ``(B_t, I)``, where ``B_t`` is the
+            mini-batch size for time ``t``. The sequences must be transposed.
+            :func:`~chainer.functions.transpose_sequence` can be used to
+            transpose a list of :func:`~chainer.Variable`\ s each representing
+            a sequence. When sequences has different lengths, they must be
+            sorted in descending order of their lengths before transposing.
+            :func:`~chainer.functions.transpose_sequence`
```
Comment (niboshi) [inline chainer/functions/connection/n_step_lstm.py]: Please remove this sentence: ``` :func:`~chainer.functions.transpose_sequence` transpose a list of :func:`~chainer.Variable` holding sequence. ```

File: chainer/functions/connection/n_step_lstm.py [hunk 32]
```diff
@@ -192,51 +237,106 @@ def n_step_bilstm(
...
```
Comment (niboshi) [inline chainer/functions/connection/n_step_lstm.py]: Please remove this sentence: ```:func:`~chainer.functions.transpose_sequence` transpose a list of :func:`~chainer.Variable` holding sequence.```

File: chainer/functions/connection/n_step_lstm.py [hunk 33]
```diff
@@ -253,52 +353,54 @@ def n_step_lstm_base(
...
```
Comment (niboshi) [inline chainer/functions/connection/n_step_lstm.py]: Please remove this sentence: ```:func:`~chainer.functions.transpose_sequence` transpose a list of :func:`~chainer.Variable` holding sequence.```

File: chainer/functions/connection/n_step_lstm.py [hunk 34]
```diff
@@ -77,55 +77,99 @@ def n_step_lstm(
...
+            transpose a list of :func:`~chainer.Variable` holding sequence.
             So ``xs`` needs to satisfy
             ``xs[t].shape[0] >= xs[t + 1].shape[0]``.
 
     Returns:
         tuple: This functions returns a tuple concaining three elements,
-            ``hy``, ``cy`` and ``ys``.
+        ``hy``, ``cy`` and ``ys``.
 
             - ``hy`` is an updated hidden states whose shape is same as ``hx``.
```
Comment (niboshi) [inline chainer/functions/connection/n_step_lstm.py]: ```is same as``` -> ```is the same as```

File: chainer/functions/connection/n_step_lstm.py [hunk 35]
```diff
@@ -77,55 +77,99 @@ def n_step_lstm(
...
             - ``cy`` is an updated cell states whose shape is same as ``cx``.
```
Comment (niboshi) [inline chainer/functions/connection/n_step_lstm.py]: ```is same as``` -> ```is the same as```

File: chainer/functions/connection/n_step_lstm.py [hunk 36]
```diff
@@ -192,51 +237,106 @@ def n_step_bilstm(
...
+            transpose a list of :func:`~chainer.Variable` holding sequence.
             So ``xs`` needs to satisfy
             ``xs[t].shape[0] >= xs[t + 1].shape[0]``.
 
     Returns:
         tuple: This functions returns a tuple concaining three elements,
-            ``hy``, ``cy`` and ``ys``.
+        ``hy``, ``cy`` and ``ys``.
 
             - ``hy`` is an updated hidden states whose shape is same as ``hx``.
```
Comment (niboshi) [inline chainer/functions/connection/n_step_lstm.py]: ```is same as``` -> ```is the same as```

File: chainer/functions/connection/n_step_lstm.py [hunk 37]
```diff
@@ -192,51 +237,106 @@ def n_step_bilstm(
...
             - ``cy`` is an updated cell states whose shape is same as ``cx``.
```
Comment (niboshi) [inline chainer/functions/connection/n_step_lstm.py]: ```is same as``` -> ```is the same as```

File: chainer/functions/connection/n_step_lstm.py [hunk 38]
```diff
@@ -253,52 +353,54 @@ def n_step_lstm_base(
...
         use_bi_direction (bool): If ``True``, this function uses Bi-directional
             LSTM.
 
     Returns:
         tuple: This functions returns a tuple concaining three elements,
-            ``hy``, ``cy`` and ``ys``.
+        ``hy``, ``cy`` and ``ys``.
 
             - ``hy`` is an updated hidden states whose shape is same as ``hx``.
             - ``cy`` is an updated cell states whose shape is same as ``cx``.
```
Comment (niboshi) [inline chainer/functions/connection/n_step_lstm.py]: ```is same as``` -> ```is the same as```

File: chainer/functions/connection/n_step_lstm.py [hunk 39]
```diff
@@ -253,52 +353,54 @@ def n_step_lstm_base(
...
             ``xs[t].shape[0] >= xs[t + 1].shape[0]``.
         use_bi_direction (bool): If ``True``, this function uses Bi-directional
             LSTM.
 
     Returns:
         tuple: This functions returns a tuple concaining three elements,
-            ``hy``, ``cy`` and ``ys``.
+        ``hy``, ``cy`` and ``ys``.
 
             - ``hy`` is an updated hidden states whose shape is same as ``hx``.
```
Comment (niboshi) [inline chainer/functions/connection/n_step_lstm.py]: ```is same as``` -> ```is the same as```

File: chainer/functions/connection/n_step_lstm.py [hunk 40]
```diff
@@ -77,55 +77,99 @@ def n_step_lstm(
...
-            mini-batch size for time ``t``, and ``I`` is size of input units.
-            Note that this functions supports variable length sequences.
-            When sequneces has different lengths, sort sequences in descending
-            order by length, and transpose the sorted sequence.
-            :func:`~chainer.functions.transpose_sequence` transpose a list
-            of :func:`~chainer.Variable` holding sequence.
+            for time ``t``. Its shape is ``(B_t, I)``, where ``B_t`` is the
+            mini-batch size for time ``t``. The sequences must be transposed.
+            :func:`~chainer.functions.transpose_sequence` can be used to
+            transpose a list of :func:`~chainer.Variable`\ s each representing
```
Comment (niboshi) [inline chainer/functions/connection/n_step_lstm.py]: Sorry, please rewrite ``` :func:`~chainer.Variable` ``` -> ``` :class:`~chainer.Variable` ```.

File: chainer/functions/connection/n_step_lstm.py [hunk 41]
```diff
@@ -192,51 +237,106 @@ def n_step_bilstm(
...
```
Comment (niboshi) [inline chainer/functions/connection/n_step_lstm.py]: Ditto. ``` :func:`~chainer.Variable` ``` -> ``` :class:`~chainer.Variable` ```.

File: chainer/functions/connection/n_step_lstm.py [hunk 42]
```diff
@@ -253,52 +353,54 @@ def n_step_lstm_base(
...
```
Comment (niboshi) [inline chainer/functions/connection/n_step_lstm.py]: Ditto. ``` :func:`~chainer.Variable` ``` -> ``` :class:`~chainer.Variable` ```.

File: chainer/functions/connection/n_step_lstm.py (hunk 40 above)
Comment (keisuke-umezawa) [reply chainer/functions/connection/n_step_lstm.py]: fixed.

File: chainer/functions/connection/n_step_lstm.py (hunk 31 above)
Comment (keisuke-umezawa) [reply chainer/functions/connection/n_step_lstm.py]: fixed.

File: chainer/functions/connection/n_step_lstm.py (hunk 35 above)
Comment (keisuke-umezawa) [reply chainer/functions/connection/n_step_lstm.py]: fixed.

File: chainer/functions/connection/n_step_lstm.py (hunk 41 above)
Comment (keisuke-umezawa) [reply chainer/functions/connection/n_step_lstm.py]: fixed.

File: chainer/functions/connection/n_step_lstm.py (hunk 32 above)
Comment (keisuke-umezawa) [reply chainer/functions/connection/n_step_lstm.py]: fixed.

File: chainer/functions/connection/n_step_lstm.py (hunk 36 above)
Comment (keisuke-umezawa) [reply chainer/functions/connection/n_step_lstm.py]: fixed.

File: chainer/functions/connection/n_step_lstm.py (hunk 37 above)
Comment (keisuke-umezawa) [reply chainer/functions/connection/n_step_lstm.py]: fixed.

File: chainer/functions/connection/n_step_lstm.py (hunk 42 above)
Comment (keisuke-umezawa) [reply chainer/functions/connection/n_step_lstm.py]: fixed.

File: chainer/functions/connection/n_step_lstm.py (hunk 33 above)
Comment (keisuke-umezawa) [reply chainer/functions/connection/n_step_lstm.py]: fixed.

File: chainer/functions/connection/n_step_lstm.py (hunk 39 above)
Comment (keisuke-umezawa) [reply chainer/functions/connection/n_step_lstm.py]: fixed.

File: chainer/functions/connection/n_step_lstm.py (hunk 38 above)
Comment (keisuke-umezawa) [reply chainer/functions/connection/n_step_lstm.py]: fixed.

**Comment by keisuke-umezawa** [comment]
@niboshi I have just fixed them!

**Comment by niboshi** [comment]
Thank you for a quick fix!

**Comment by niboshi** [comment]
jenkins, test this please

**Comment by niboshi** [comment]
LGTM. Thanks!

**Pull request merged**